### PR TITLE
578

### PR DESCRIPTION
**Pull request by KaarelKa**

### CODE REVIEW

**Comment by KaarelKa** [comment]
Also contains changes for [454](https://github.com/keeleinstituut/tv-tolkevarav/issues/454) task

File: src/hooks/requests/useVolumes.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  const { mutateAsync: removeAssignmentVolume, isLoading } = useMutation({
+    mutationKey: ['subprojects', id],
+    mutationFn: (payload: { volumeId?: string }) =>
+      apiClient.delete(`${endpoints.VOLUMES}/${payload.volumeId}`),
+    onSuccess: (_, { volumeId }) => {
+      queryClient.setQueryData(
+        ['subprojects', id],
+        (oldData?: SubProjectResponse) => {
+          const { data: previousData } = oldData || {}
+          if (!previousData) return oldData
```
Comment (rebekkabb) [inline src/hooks/requests/useVolumes.ts]: Not a blocker but maybe also extract the logic here into a separate method like getNewSubProjectWithAssignment

**Review by rebekkabb — APPROVED** [review]
Left one comment, but the modal seems to work and the other stuff lgtm

File: src/hooks/requests/useVolumes.ts (hunk 1 above)
Comment (KaarelKa) [reply src/hooks/requests/useVolumes.ts]: Yeah not a bad idea, I usually do it once I have at least 2 places where I need to use something, but could be done before here as well

**Pull request merged**